web-app-router/components/DecoratedButton/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button, CommonButtonProps } from "@/components/Button";
+import { memo } from "react";
+import { Icon, IconType } from "@/components/Icon";
+import clsx from "clsx";
+
+type DecoratedButtonProps = CommonButtonProps & {
+  icon?: IconType;
+  showArrowRight?: boolean;
+  disabled?: boolean;
+  loading?: boolean;
+  className?: string;
+  variant?: "primary" | "secondary" | "danger";
+};
+
+export const DecoratedButton = memo(function DecoratedButton(
+  props: DecoratedButtonProps
+) {
+  const {
+    icon,
+    showArrowRight,
+    loading,
+    disabled,
+    className,
+    variant = "primary",
+    ...restProps
+  } = props;
+  const buttonStyles = {
```

Comment:
@andy-t-wang Lets divide multiline code with empty row.

```
} = props

const buttonStyles = {
```

Same for the row before return